### ISSUE TITLE
Add `--insecure` flag that allows users to access the tunnel without verifying certificates

### ISSUE THREAD

**Issue by yinonavraham** [coder/coder#5085]
### What happened?

I'm trying to run Coder in Docker on my local MacOS, following the [instructions](https://coder.com/docs/coder-oss/latest/quickstart/docker) in the documentation, using:
```
export CODER_DATA=$HOME/.config/coderv2-docker
export DOCKER_GROUP=$(getent group docker | cut -d: -f3)
mkdir -p $CODER_DATA
docker run --rm -it \
    -v $CODER_DATA:/home/coder/.config \
    -v /var/run/docker.sock:/var/run/docker.sock \
    --group-add $DOCKER_GROUP \
    ghcr.io/coder/coder:latest
```

**The first issue:**
```
getent: command not found
```
Using `export DOCKER_GROUP=$(stat -f '%g' $(which docker))` instead seems to be ok (but I'm not sure it is the best approach).

**The second issue**, and here I'm currently stuck, is that the container fails to start. 
It fails with the error message: `Post "https://pit-1.try.coder.app/tun": x509: certificate signed by unknown authority`.
This is the output:
```
Coder v0.12.7+49b340e - Remote development on your infrastucture
Using built-in PostgreSQL (/home/coder/.config/coderv2/postgres)
Opening tunnel so workspaces can connect to your deployment. For production scenarios, specify an external access URL
Stopping built-in PostgreSQL...
Stopped built-in PostgreSQL
create tunnel: start update routine: send config to server: do request: Post "https://pit-1.try.coder.app/tun": x509: certificate signed by unknown authority
Run 'coder server --help' for usage.
```
I tried also using [`docker-compose.yaml`](https://github.com/coder/coder/blob/main/docker-compose.yaml), but got similar results:
```
coder-coder-1     | Coder v0.12.7+49b340e - Remote development on your infrastucture
coder-coder-1     | Opening tunnel so workspaces can connect to your deployment. For production scenarios, specify an external access URL
coder-coder-1     | Using tunnel in US East Pittsburgh with latency  0s .
coder-coder-1     | create tunnel: start update routine: send config to server: do request: Post "https://pit-1.try.coder.app/tun": x509: certificate signed by unknown authority
coder-coder-1     | Run 'coder server --help' for usage.
coder-coder-1 exited with code 1
```

My OS Details:
```
MacBook Pro
macOS Monterey v12.5.1
Apple M1 Max
```

Docker Version:
```
Client:
 Cloud integration: v1.0.29
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:04:45 2022
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.12.0 (85629)
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:01 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
``` 

### What did you expect to happen?

I expected coder server to simply start... 
Or give me a way to mitigate such an error (e.g. such as `curl --insecure`).

**Comment by kylecarbs** [comment]
@yinonavraham we can certainly simplify this to hopefully remove the `getent` requirement.

I'm not sure why the certificate failure occurred... I'd recommend installing Coder on the host machine while we fix this.

**Comment by yinonavraham** [comment]
Thanks, this is exactly what I did - running locally for now.

**Comment by yinonavraham** [comment]
OK, I found the cause for the certificate issue...
We have an internet security agent installed which acts like a trusted man-in-the-middle. It uses its own certificate, signed by their own root CA. But since this root CA is not available in the docker image, the certificate is not trusted...

To resolve it I simply disabled this feature and coder now starts successfully in Docker.

BTW, I had to go back and solve it, because starting a workspace in Docker also didn't work. Looking in the container's log:
```
curl -fsSL --compressed https://xxx.pit-1.try.coder.app/bin/coder-linux-arm64 -o coder
curl: (60) SSL certificate problem: unable to get local issuer certificate
```

But I would still expect coder to provide a way to mitigate it.

Thanks!
